Library/Homebrew/brew.sh

```diff
@@ -12,7 +12,19 @@ then
   then
     export LC_ALL="en_US.UTF-8"
   else
-    export LC_ALL="C.UTF-8"
+    utf8_locales=$(locale -a | grep --extended-regexp '\.(utf|UTF-)8')
+    re1='\bC\.(utf|UTF-)8'
+    re2='\ben_US\.(utf|UTF-)8'
+    re3='\b[a-z][a-z]_[A-Z][A-Z]\.(utf|UTF-)8'
+    if [[ $utf8_locales =~ $re1 || $utf8_locales =~ $re2 || $utf8_locales =~ $re3 ]]
+    then
+      export LC_ALL=${BASH_REMATCH[0]}
+    elif locale -a | grep --quiet --word-regexp C
+    then
+      export LC_ALL=C
+    else
+      echo "Warning: Could not find usable locale." >&2
```

Comment:
What's the action item for the user here?

Reply:
(I'm wondering if this should be not output or be a hard-stop error).

Reply:
install either a UTF-8 locale of choice or C locale.

Reply:
In that case, yeh, I'd make this an `odie` with instructions on how to resolve.

Reply:
sounds good to me but I'd like to run this by @sjackman  👀

Reply:
The user is pretty much guaranteed to have a `C` locale. If they do not, the world is burning. It's more likely that `locale` failed to run (they don't have this executable in their `PATH`, which itself is pretty unlikely) than they don't have a `C` locale. Rather than emit a warning or error in this `else` case, I'd just set `LC_ALL=C` and move on.

Reply:
> Rather than emit a warning or error in this `else` case, I'd just set `LC_ALL=C` and move on.

OK

Reply:
Shall we remove `c_regex` then?

Reply:
Makes sense.